protocol/codec.go

```diff
@@ -132,13 +155,49 @@ func EncodeJSON(obj interface{}) []byte {
 	return b
 }
 
-// Decode attempts to decode a msgpack-encoded byte buffer into an
-// object instance pointed to by objptr
-func Decode(b []byte, objptr interface{}) error {
+// DecodeReflect attempts to decode a msgpack-encoded byte buffer
+// into an object instance pointed to by objptr, using reflection.
+func DecodeReflect(b []byte, objptr interface{}) error {
 	dec := codec.NewDecoderBytes(b, CodecHandle)
 	return dec.Decode(objptr)
 }
 
+// DecodeMsgp attempts to decode a msgpack-encoded byte buffer into
+// an object instance pointed to by objptr, requiring that we pre-
+// generated the code for doing so using msgp.
+func DecodeMsgp(b []byte, objptr msgp.Unmarshaler) (err error) {
+	defer func() {
+		if x := recover(); x != nil {
+			err = fmt.Errorf("DecodeMsgp: %v", x)
+		}
+	}()
+
+	var rem []byte
+	rem, err = objptr.UnmarshalMsg(b)
+	if err != nil {
+		return err
+	}
+
+	// go-codec compat: allow remaining bytes, because go-codec allows it too
+	if false {
```

Comment:
dead code

Reply:
Do you think we should remove it?  I have hopes that we'll eventually grep for these `go-codec compat` places and make our decoder more strict, and this is one such case.

Reply:
alright, fine with me. just thought this might have been accidental